Makefile

```diff
@@ -247,11 +247,19 @@ ide: zephyr
 .PHONY: ashell
 ashell: zephyr
 
-# Flash Arduino 101 x86 and arc images
+# Flash for Arduino 101 x86 and arc images, olimex_stm32_e407, 96b_carbon
 .PHONY: dfu
 dfu:
+ifeq ($(BOARD), arduino_101)
 	dfu-util -a x86_app -D $(A101BIN).dfu
 	dfu-util -a sensor_core -D $(A101SSBIN).dfu
+endif
+ifeq ($(BOARD), olimex_stm32_e407)
+	dfu-util -a 0 -d 0483:df11 -D $(OUT)/olimex_stm32_e407/zephyr.bin --dfuse-address 0x08000000
+endif
+ifeq ($(BOARD), 96b_carbon)
+	dfu-util -a 0 -d 0483:df11 -D $(OUT)/96b_carbon/zephyr.bin --dfuse-address 0x08000000
+endif
```

Comment:
I figured it out that `make BOARD=<BOARD> flash` is working very well with any supported boards. So, wondering why we need to handle these board specific flashing commands (except A101) here. I tested with olimex_stm32_e407, stm32f4_disco and 96b_carbon, and no issues flashing those boards using `make BOARD=<BOARD> flash`.

Reply:
Yes, make flash will work, like I've pointed out to @grgustaf that we might want to switch from make dfu to make flash for a101 as well, since it would wait for you to press reset instead of you having to flash within 5 secs. Since a101 flashes two images all at once, might be little tricky to get it working